hypervisor/arch/x86/guest/vm.c

```diff
@@ -621,17 +629,39 @@ void start_vm(struct acrn_vm *vm)
 int32_t reset_vm(struct acrn_vm *vm)
 {
 	uint16_t i;
+	uint16_t this_pcpu_id = get_pcpu_id();
+	bool offline_pcpu;
+	uint64_t mask = 0UL;
 	struct acrn_vcpu *vcpu = NULL;
-	int32_t ret;
+	int32_t ret = 0;
 
 	if (vm->state == VM_PAUSED) {
 		foreach_vcpu(i, vm, vcpu) {
+			offline_pcpu = is_lapic_pt_enabled(vcpu);
 			reset_vcpu(vcpu);
+
+			if (offline_pcpu) {
+				if (this_pcpu_id != vcpu->pcpu_id) {
+					bitmap_set_nolock(vcpu->pcpu_id, &mask);
+					make_pcpu_offline(vcpu->pcpu_id);
+				} else {
+					pr_warn("%s: cannot offline self(%u)",
+						__func__, this_pcpu_id);
+					ret = -EINVAL;
+				}
+			}
+		}
+
+		wait_pcpus_offline(mask);
+
+		if (mask != 0UL && !start_pcpus(mask)) {
```

Comment:
please fix it by add brackets in another patch.
"**if ((mask != 0UL) && !start_pcpus(mask))** {"

Reply:
#3716